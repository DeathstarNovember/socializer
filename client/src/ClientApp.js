import React, { useRef } from "react";
import { ApolloProvider } from "@apollo/react-hooks";
import { BrowserRouter } from "react-router-dom";
import { createClient } from "util/apollo";
import App from "./App";

const ClientApp = () => {
  const client = useRef(createClient());

  return (
    <ApolloProvider client={client.current.client}>
      <BrowserRouter>
        <App socket={client.current.absintheSocket} />
      </BrowserRouter>
    </ApolloProvider>
  );
};

export default ClientApp;
