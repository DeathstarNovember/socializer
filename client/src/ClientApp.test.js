import React from "react";
import { render } from "@testing-library/react";
import ClientApp from "./ClientApp";

describe("ClientApp", () => {
  it("renders correctly", () => {
    const { container } = render(<ClientApp />);
    expect(container).toMatchSnapshot();
  });
});
