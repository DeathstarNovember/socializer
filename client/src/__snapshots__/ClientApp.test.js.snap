// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClientApp renders correctly 1`] = `
<div>
  <nav
    class="mb-4 navbar navbar-expand navbar-dark bg-primary fixed-top"
  >
    <div
      class="container"
    >
      <a
        class="navbar-brand"
        href="/"
      >
        Socializer
      </a>
      <button
        aria-label="Toggle navigation"
        class="navbar-toggler"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="justify-content-end navbar-collapse collapse"
      >
        <div
          class="navbar-nav"
        >
          <a
            class="nav-link"
            href="/login"
            role="button"
          >
            Log in
          </a>
          <a
            class="nav-link"
            href="/signup"
            role="button"
          >
            Sign up
          </a>
        </div>
      </div>
    </div>
  </nav>
  <div
    class="container"
  >
    <div
      class="row"
    >
      <div
        class="d-none d-md-block col-md-4"
      >
        <div
          class="p-2 empty-state"
        >
          <h4>
            Chat
          </h4>
          <p>
            <a
              href="/login"
            >
              Log in
            </a>
             to chat
          </p>
        </div>
      </div>
      <div
        class="col-md-8 col-12"
      >
        <h4>
          Feed
        </h4>
        <div
          class="d-flex justify-content-center"
        >
          <div
            class="spinner-border"
            role="status"
          >
            <span
              class="sr-only"
            >
              Loading...
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
